pkg/app/app.go

```diff
@@ -151,6 +151,29 @@ func isDirectoryAGitRepository(dir string) (bool, error) {
 	return info != nil, err
 }
 
+func isBareRepo(osCommand *oscommands.OSCommand) (bool, error) {
+	res, err := osCommand.Cmd.New("git rev-parse --is-bare-repository").DontLog().RunWithOutput()
```

Comment:
@jesseduffield should we somehow conflate this `isBareRepo` and [this](https://github.com/nullishamy/lazygit/blob/bdb0b9ae6e84ce7261190f156f8f16fc7d62cffd/pkg/gui/recent_repos_panel.go#L156) one?

Currently I think the one from `recent_repos_panel` isn't reachable from `app.go`, but maybe it's better to make it public rather than have a duplicate?

Reply:
Good idea, let's make the existing one public like you say

Reply:
How would that look, and do you want that implemented in this PR? Also how do I go about accessing the existing function?

Reply:
> How would that look, and do you want that implemented in this PR? 

It seems like a trivial change, I see no reason why to postpone it.

> Also how do I go about accessing the existing function?

Should be accessible through `app.IsBareRepo` once you make it public (capitalize it)

Reply:
I'll give it a go.

Reply:
Whoops, I mixed them up, the one from `recent_repos_panel` should be public.

Reply:
I'm not sure I understand. The existing method is not associated in any way to the `App` type? If we make the git commands public on the `Gui` type, we can call `app.Gui.Git.Status.IsBareRepo()` though that doesnt seem to be what was suggested.

Reply:
My bad, I mixed the two functions up, I thought the one from `app` needed to be made public, but it's the other way around. But then I read that Jesse said:
> Good idea, let's make the existing one public like you say

Which means the one from [status.go](https://github.com/jesseduffield/lazygit/blob/70a46028e1499d01d474212c5be0167d41cf5a81/pkg/commands/git_commands/status.go#L57) should be made public.

_*HOWEVER*_, seeing as it is a member function of `StatusCommands`, and it depends on `gogit`, I propose we indeed do make _your_ newly added function the only one which can then be invoked with `app.IsBareRepo(path)`. I am not sure where's the best place to put such helper functions since there are others, but maybe Jesse knows some place. In the meantime, I propose you leave it where it is, in `app.go`

Reply:
Do we use go-git in other places? If so, I don't see a reason to drop it. Unless there's some other compelling reason.

Reply:
> Which means the one from [status.go](https://github.com/jesseduffield/lazygit/blob/70a46028e1499d01d474212c5be0167d41cf5a81/pkg/commands/git_commands/status.go#L57) should be made public.

Unless I'm being stupid, it already is?

Reply:
> Do we use go-git in other places?

Yes.

> If so, I don't see a reason to drop it. Unless there's some other compelling reason.

I'm not sure if the `StatusCommands` exists at the point in time when we want to check if the current repo is bare. If we refactor the current method to be a free function that takes a `Worktree` (because that's what the current method requires), the caller needs to create it, and it looks to me way too convoluted just to check if the repo is bare.

Reply:
> Unless I'm being stupid, it already is?

You're not being stupid, and it isn't already public. Functions are made public (exported) by having a name starting with an uppercase letter.

And the current `IsBareRepo` from `status.go` then needs to be refactored so it invokes `app.IsBareRepo`.

Reply:
So, attach the function to `App`, find a way to pass `App` into the `StatusCommands`, and refactor the function to use the app function?

Reply:
We cant import App into status.go because its circular :^)

Reply:
I can't say I know of a better way to do this, without duplication.

Reply:
I say we wait for Jesse to see if he can `deus ex machina` this pickle.

_I also say we extract it to something like `pkg/commands/git_commands/misc.go`, throw in `IsDirectoryAGitRepository` as well and call it a day._

We'll see!

Reply:
I meaaaaaan that's not even a bad idea per-se. But we can wait for Jesse to comment on this one. Go is a real pain sometimes.

Reply:
the app package can depend on the commands package (which contains git.go) but not vice-versa. I see two approaches we can take here:
1) make it so that we can instantiate the git struct (in git.go) without having to do the validation stuff that currently happens there.
2) Move the code we've got here into `pkg/commands/git_commands/status.go` Like so:
```go
func (self *StatusCommands) IsBareRepo() (bool, error) {
	return IsBareRepo(self.os)
}

func IsBareRepo(osCommand *oscommands.OSCommand) (bool, error) {
	res, err := osCommand.Cmd.New("git rev-parse --is-bare-repository").DontLog().RunWithOutput()
	if err != nil {
		return false, err
	}

	// The command returns output with a newline, so we need to strip
	return strconv.ParseBool(strings.TrimSpace(res))
}
```
Then in app.go we call IsBareRepo, passing in our os struct.

I would prefer option 1 but that deserves its own PR, so I'm happy to go with option 2 for now.

Reply:
the go-git method is very fast btw because it's just checking if a value is nil, but this function is only called when you want to view recent repos so it's not a hot path.

Reply:
That should be that sorted, took me a while to figure out how to import the function lol.